src/material/checkbox/checkbox.ts

```diff
@@ -214,6 +231,9 @@ export class MatCheckbox extends _MatCheckboxMixinBase implements ControlValueAc
         });
       }
     });
+
+    // TODO: Remove this after the `_clickAction` parameter is removed as an injection parameter.
+    this._clickAction = this._clickAction || this._options.clickAction;
```

Comment:
Since `clickAction` is deprecated, I think `options.clickAction` should take precedence

Reply:
Discussed in person - I think the way it is captures the most common use cases, especially since the options object is provided in root and always defined, so it's not easy to tell if the user is switched to it